cl/search/tests.py

```diff
@@ -334,6 +334,75 @@ def test_issue_729_url_coalescing(self) -> None:
         RECAPDocument.objects.all().delete()
 
 
+class AdvancedTest(IndexedSolrTestCase):
+    """
+    Advanced query techniques
+    """
+
+    def setUp(self) -> None:
+        self.fixtures.append("test_objects_search.json")
+
+        super(AdvancedTest, self).setUp()
+
+    def test_a_intersection_query(self) -> None:
+        """Does AND queries work"""
+        r = self.client.get(reverse("show_results"), {"q": "Howard AND Honda"})
+        self.assertIn("Howard", r.content.decode())
+        self.assertIn("Honda", r.content.decode())
+        self.assertIn("1 Opinion", r.content.decode())
+
+    def test_a_union_query(self) -> None:
+        """Does OR queries work"""
+        r = self.client.get(
+            reverse("show_results"), {"q": "Howard OR Lissner"}
+        )
+        self.assertIn("Howard", r.content.decode())
+        self.assertIn("Lissner", r.content.decode())
+        self.assertIn("2 Opinions", r.content.decode())
+
+    def test_query_negation(self) -> None:
+        """Does negation query work"""
+        r = self.client.get(reverse("show_results"), {"q": "Howard"})
+        self.assertIn("Howard", r.content.decode())
+        self.assertIn("1 Opinion", r.content.decode())
+
+        r = self.client.get(reverse("show_results"), {"q": "Howard NOT Honda"})
+        self.assertIn("had no results", r.content.decode())
+
+        r = self.client.get(reverse("show_results"), {"q": "Howard !Honda"})
+        self.assertIn("had no results", r.content.decode())
+
+        r = self.client.get(reverse("show_results"), {"q": "Howard -Honda"})
+        self.assertIn("had no results", r.content.decode())
+
+    def test_query_phrase(self) -> None:
+        """Can we query by phrase"""
+        r = self.client.get(
+            reverse("show_results"), {"q": "Testing Supreme Court"}
```

Comment:
This doesn't quite look like a phrase search to me. Shouldn't it have encoded quotes or something like that?

Reply:
Sure I can add the others. Wild card, date and range tests already exist. I thought there was a proximity ('"net neutrality"~2', '"net neutrality"~2') test but I can add something more specific. I believe the only missing tests are related and fielded.

That said, I will just update AdvancedTest with all of the operations represented here to make things more concise.

https://www.courtlistener.com/help/search-operators/

Reply:
That's great, thanks Marion. When it's ready, I think you can use the "Request Review" feature again or flag me to look again and I'll take another look.